src/wx/vbam-options-static.cpp

```diff
@@ -0,0 +1,830 @@
+    kHQ2x,
+    kLQ2x,
+    kSimple2x,
+    kSimple3x,
+    kHQ3x,
+    kSimple4x,
+    kHQ4x,
+    kXbrz2x,
+    kXbrz3x,
+    kXbrz4x,
+    kXbrz5x,
+    kXbrz6x,
+    kPlugin, // This must always be last.
+
+    // Do not add anything under here.
+    kLast,
+};
+constexpr size_t kNbFilterFunctions = static_cast<size_t>(FilterFunction::kLast);
+
+// These MUST follow the same order as the definitions of the enum above.
+// Adding an option without adding to this array will result in a compiler
+// error since kNbFilterFunctions is automatically updated.
+static const std::array<wxString, kNbFilterFunctions> kFilterStrings = {
+    "none",
+    "2xsai",
+    "super2xsai",
+    "supereagle",
+    "pixelate",
+    "advmame",
+    "bilinear",
+    "bilinearplus",
+    "scanlines",
+    "tvmode",
+    "hq2x",
+    "lq2x",
+    "simple2x",
+    "simple3x",
+    "hq3x",
+    "simple4x",
+    "hq4x",
+    "xbrz2x",
+    "xbrz3x",
+    "xbrz4x",
+    "xbrz5x",
+    "xbrz6x",
+    "plugin",
+};
+
+// This enum must be kept in sync with the one in wxvbam.h
+enum class Interframe {
+    kNone = 0,
+    kSmart,
+    kMotionBlur,
+
+    // Do not add anything under here.
+    kLast,
+};
+constexpr size_t kNbInterframes = static_cast<size_t>(Interframe::kLast);
+
+// These MUST follow the same order as the definitions of the enum above.
+// Adding an option without adding to this array will result in a compiler
+// error since kNbInterframes is automatically updated.
+static const std::array<wxString, kNbInterframes> kInterframeStrings = {
+    "none",
+    "smart",
+    "motionblur",
+};
+
+// This enum must be kept in sync with the one in wxvbam.h
+enum class RenderMethod {
+    kSimple = 0,
+    kOpenGL,
+#ifdef __WXMSW__
+    kDirect3d,
+#elif defined(__WXMAC__)
+    kQuartz2d,
+#endif
+
+    // Do not add anything under here.
+    kLast,
+};
+constexpr size_t kNbRenderMethods = static_cast<size_t>(RenderMethod::kLast);
+
+// These MUST follow the same order as the definitions of the enum above.
+// Adding an option without adding to this array will result in a compiler
+// error since kNbRenderMethods is automatically updated.
+static const std::array<wxString, kNbRenderMethods> kRenderMethodStrings = {
+    "simple",
+    "opengl",
+#ifdef __WXMSW__
+    "direct3d",
+#elif defined(__WXMAC__)
+    "quartz2d",
+#endif
+};
+
+// This enum must be kept in sync with the one in wxvbam.h
+enum class AudioApi {
+    kSdl = 0,
+    kOpenAL,
+    kDirectSound,
+    kXAudio2,
+    kFaudio,
+
+    // Do not add anything under here.
+    kLast,
+};
+constexpr size_t kNbAudioApis = static_cast<size_t>(AudioApi::kLast);
+
+// These MUST follow the same order as the definitions of the enum above.
+// Adding an option without adding to this array will result in a compiler
+// error since kNbAudioApis is automatically updated.
+static const std::array<wxString, kNbAudioApis> kAudioApiStrings = {
+    "sdl",
+    "openal",
+    "directsound",
+    "xaudio2",
```

Comment:
directsound and xaudio2 are for Windows only, don't know if that matters for your implementation.